AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Core/SQLiteStorageEngineAdapterTests.swift

```diff
@@ -375,4 +375,120 @@ class SQLiteStorageEngineAdapterTests: BaseDataStoreTests {
             XCTFail(String(describing: error))
         }
     }
+
+    func testClearIfNewVersionShouldStoreInitialVersion() {
+
+        guard let userDefaults = UserDefaults.init(suiteName: "testVersionIsEmpty") else {
+            XCTFail("Could not create a UserDafult with this suite name")
+            return
+        }
+        userDefaults.removeObject(forKey: SQLiteStorageEngineAdapter.dbVersionKey)
+
+        let newVersion = "newVersion"
+
+        let mockFileManager = MockFileManager()
+
+        do {
+            try SQLiteStorageEngineAdapter.clearIfNewVersion(version: newVersion,
+                                                             dbFilePath: URL(string: "dbFilePath")!,
+                                                             userDefaults: userDefaults,
+                                                             fileManager: mockFileManager)
+        } catch {
+            XCTFail("Test has failed due to \(error)")
+        }
+
+        _ = UserDefaults.removeObject(userDefaults)
+    }
+
+    func testVersionInUserDefaultsRemainsSameAndFileExists() {
```

Comment:
test name like `testClearIfNewVersionWithVersionSameAsPrevious`

```
let previousVersion = "previousVersion"
userDefaults.set(previousVersion ..

try SQLiteStorageEngineAdaapter.clearIfNewVersion(previousVersion,

```

Reply:
Actually, here is having an additional var necessary?
`userDefaults.set("previousVersion", forKey: SQLiteStorageEngineAdapter.dbVersionKey)` is OK? Because "previousVersion" is used only once.